### PR TITLE
[CLI] Accept `spaces/user/repo` as repo ID prefix shorthand

### PR DESCRIPTION
**Pull request by Wauplin**
## Summary

Accept repo IDs like `spaces/user/repo`, `datasets/user/repo`, `models/user/repo` as shorthand for `user/repo --type space` etc. This is implemented at the `HFCliTyperGroup.resolve_command()` level, so it works **automatically for all commands that accept `--type`/`--repo-type`** — no per-command changes needed.

Follows the same pattern as the `--json` → `--format json` rewrite from #3919.

```bash
# before
hf download user/my-space --type space
hf discussions list user/my-dataset --type dataset

# after
hf download spaces/user/my-space
hf discussions list datasets/user/my-dataset
```

- Error if both a prefix and explicit `--type` flag are provided (ambiguous)
- Error if multiple prefixed args disagree on the type
- Commands without `--type` option are unaffected

Addresses https://github.com/huggingface/huggingface_hub/pull/3906#issuecomment-4055813007

## Test plan

- [x] `spaces/user/repo` rewrites to `user/repo --type space` on `download`
- [x] `datasets/user/repo` rewrites to `user/repo --type dataset` on `download`
- [x] `models/user/repo` rewrites to `user/repo --type model` on `download`
- [x] Error when prefix + explicit `--type` both provided
- [x] Error when prefix + explicit `--repo-type` both provided
- [x] Normal repo IDs (no prefix) unchanged
- [x] Works on nested subcommands (`repos settings`)
- [x] Not applied on commands without `--type`
- [x] Existing `--json` shorthand tests still pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Changes global CLI argument rewriting in `HFCliTyperGroup.resolve_command`, which can affect many commands’ parsing/behavior if edge cases are missed, though it’s guarded to only run on commands with `--type` and specific repo-id positionals.
> 
> **Overview**
> Adds a new CLI shorthand allowing prefixed repo IDs like `spaces/user/repo`, `datasets/user/repo`, and `models/user/repo` to be automatically rewritten to `user/repo --type {space|dataset|model}` for any command that supports `--type`/`--repo-type`.
> 
> This introduces centralized parsing in `HFCliTyperGroup` with validation (errors on conflicting prefixes or when combined with explicit `--type`/`--repo-type`, stops at `--`, and avoids rewriting non-repo positional args like filenames), updates `RepoIdArg`/generated CLI docs to mention the new format, and adds test coverage for the rewrite and edge cases (nested commands, duplicate/move, passthrough commands, and filename safety).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c0013e6a56bc8b39cfb4db2d48befe988601ac76. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/huggingface_hub/cli/_cli_utils.py [hunk 1]
```diff
@@ -97,25 +104,87 @@ class HFCliTyperGroup(typer.core.TyperGroup):
...
+            return
+
+        # Collect all positional args that look like prefixed repo IDs.
+        prefix_matches: list[tuple[int, list[str]]] = []
+        for i, arg in enumerate(args):
+            if i == 0 or arg.startswith("-"):
+                continue
+            parts = arg.split("/", 2)
+            if len(parts) == 3 and parts[0] in _REPO_TYPE_PREFIXES:
+                prefix_matches.append((i, parts))
```
[the remark this answers is not in the document]
Comment (hanouticelina) [reply src/huggingface_hub/cli/_cli_utils.py]: agree with cursor here, we should scan only repo id positional args.

File: src/huggingface_hub/cli/_cli_utils.py [hunk 2]
```diff
@@ -88,6 +88,13 @@ def _format_epilog_no_indent(epilog: Optional[str], ctx: click.Context, formatte
 
 _ALIAS_SPLIT = re.compile(r"\s*\|\s*")
 
+# Mapping from URL-style plural prefixes to repo type values used by the CLI.
+_REPO_TYPE_PREFIXES = {
+    "spaces": "space",
+    "datasets": "dataset",
+    "models": "model",
+}
```
Comment (hanouticelina) [inline src/huggingface_hub/cli/_cli_utils.py]: we can use `constants.REPO_TYPES_MAPPING` here

File: src/huggingface_hub/cli/_cli_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanouticelina) [reply src/huggingface_hub/cli/_cli_utils.py]: Codex suggested to use `cmd.params` to know which tokens are `click.Arguments`, and only inspect arguments whose name is `repo_id`

[the remark this answers is not in the document]
Comment (Wauplin) [reply src/huggingface_hub/cli/_cli_utils.py]: addressed in https://github.com/huggingface/huggingface_hub/pull/3929/commits/24a369f97f174b8c1734426145f0be0a0b42404d

File: src/huggingface_hub/cli/_cli_utils.py [hunk 3]
```diff
@@ -98,25 +98,117 @@ class HFCliTyperGroup(typer.core.TyperGroup):
...
+
+        # Locate the `repo_id` positional argument and its index among Arguments.
+        repo_id_pos = next(
+            (
+                idx
+                for idx, param in enumerate(param for param in cmd.params if isinstance(param, click.Argument))
+                if param.name == "repo_id"
+            ),
+            None,
+        )
```
[the remark this answers is not in the document]
Comment (Wauplin) [reply src/huggingface_hub/cli/_cli_utils.py]: addressed in https://github.com/huggingface/huggingface_hub/pull/3929/commits/b49a822229a25c1321de92715fb5c13c96a7c967

**Comment by Wauplin** [comment]
Should be good now :)
